fetch_treaties.py

```diff
@@ -0,0 +1,75 @@
+"""
+Chron.
+Copyright (C) 2018 Alisa Belyaeva, Ata Ali Kilicli, Amaury Martiny,
+Daniil Mordasov, Liam O'Flynn, Mikhail Orlov.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from datetime import datetime
+import requests
+import os
+
+URL = "https://query.wikidata.org/sparql"
+QUERY = """
+SELECT ?treaty ?treatyLabel ?time ?location
+WHERE 
+{ 
+  ?treaty wdt:P31 wd:Q625298.
+  ?treaty wdt:P585 ?time.
+  ?treaty wdt:P276 ?location.
+  FILTER (?time > "1789-01-01T00:00:00Z"^^xsd:dateTime)
+  FILTER (?time < "1815-01-01T00:00:00Z"^^xsd:dateTime)
+  SERVICE wikibase:label { bd:serviceParam wikibase:language "en" }
+}
+"""
+R_TREATIES = requests.get(URL, params={"format": "json", "query": QUERY})
+TREATIES = R_TREATIES.json()
+
+for treaty in TREATIES["results"]["bindings"]:
```

Comment:
Instead of doing n more requests, you could actually do one sparql request to get location + coords:
```
SELECT ?treaty ?treatyLabel ?time ?location ?coors
WHERE 
{ 
  ?treaty wdt:P31 wd:Q625298.
  ?treaty wdt:P585 ?time.
  ?treaty wdt:P276 ?location.
  ?location wdt:P625 ?coors
  FILTER (?time > "1789-01-01T00:00:00Z"^^xsd:dateTime)
  FILTER (?time < "1816-01-01T00:00:00Z"^^xsd:dateTime)
  SERVICE wikibase:label { bd:serviceParam wikibase:language "en" }
}
```

sorry, my fault, should have put this version in the excel file.

Reply:
Thanks, I was experimenting with this but when I tried to return P625 only one or two had a value for it, which is why I did the second query based on P276. It seems like most of them have a P625 though now that I look

Reply:
When I do `OPTIONAL {?treaty wdt:P276 ?location }.`, I get 28 results, which means some treaties don't have location at all. I was planning to add them manually on wikidata.